src/Diagnostics.RuntimeHost/Controllers/ControllerBase.cs

```diff
@@ -85,12 +85,31 @@ protected OperationContext<TResource> PrepareContext<TResource>(TResource resour
             };
 
             var response = (Response)await invoker.Invoke(new object[] { dataProviders, context, res });
+
+            var dataProvidersMetadata = GetDataProvidersMetadata(dataProviders);
             response.UpdateDetectorStatusFromInsights();
-            var apiResponse = DiagnosticApiResponse.FromCsxResponse(response);
+            var apiResponse = DiagnosticApiResponse.FromCsxResponse(response, dataProvidersMetadata);
```

Comment:
make sure to not add the query information for external calls. Operation Context has a member (IsExternal)

Reply:
done